src/Decompiler/Loading/Loader.cs

```diff
@@ -298,6 +309,30 @@ public Program LoadRawImage(ImageLocation imageLocation, byte[] image, Address?
             return program;
         }
 
+        private byte[]? ExtractRawFileFromArchives(ImageLocation imageLocation, byte[] image, Address? addrLoad)
+        {
+            var fsLocation = ImageLocation.FromUri(imageLocation.FilesystemPath);
+            var binaryImage = LoadBinaryImage(fsLocation, image, null, addrLoad);
+            for (int i = 0; i < imageLocation.Fragments.Length; ++i)
+            {
+                var fragment = imageLocation.Fragments[i];
+                if (binaryImage is not IArchive archive)
+                    throw new InvalidOperationException($"Image '{imageLocation}' expects an archive or file container, but none was loaded.");
+                var item = archive[fragment];
+                if (item is not ArchivedFile file)
+                    throw new InvalidOperationException($"Fragment '{fragment}' does not refer to an archived file.");
+                binaryImage = file.LoadImage(Services, addrLoad);
+                if (binaryImage is Blob blob)
+                {
+                    if (i == imageLocation.Fragments.Length - 1)
+                        return blob.Image;
+                    // Archive doesn't know this format, perhaps Reko does.
+                    binaryImage = LoadBinaryImage(blob.Location, blob.Image, null, addrLoad);
+                }
+            }
+            return image;
```

Comment:
We shouldn't reach this point duiring correct execution. Throwing exception is better than returning incorrect `image`

Reply:
OK. Done.